File: Second Semester(Java and C++)/List 3/C++/Pentagon.cpp
#include "Pentagon.hpp"

#include <cmath>

Pentagon::Pentagon(int side) noexcept(true) {
    this->side = side;
}

double Pentagon::getArea() noexcept(true) {
    return sqrt(25 + 10 * sqrt(5)) * ((side * side) / 4);
}

double Pentagon::getPerimeter() noexcept(true) {
    return 5 * side;
}

std::string Pentagon::getName() noexcept(true) {
    return "Pentagon";
}

File: Second Semester(Java and C++)/List 3/Java/Task 1/Pentagon.java
public class Pentagon extends Shape {
    private double side;

    public Pentagon(int side) {
        this.side = side;
    }

    @Override
    public double getArea() {
        return Math.sqrt(25 + 10 * Math.sqrt(5)) * (Math.pow(side, 2) / 4);
    }

    @Override
    public double getPerimeter() {
        return 5 * side;
    }

    @Override
    public String getName() {
        return "Pentagon";
    }
}
